waggle-dance-integration-tests/src/test/java/com/hotels/bdp/waggledance/WaggleDanceIntegrationTest.java

```diff
@@ -202,6 +209,36 @@ public void typical() throws Exception {
     // Remote table
     String waggledRemoteDbName = REMOTE_DATABASE;
     assertTypicalRemoteTable(proxy, waggledRemoteDbName);
+
+  }
+
+  @Test
+  public void typicalGetAllFunctions() throws Exception {
+    runner = WaggleDanceRunner
+        .builder(configLocation)
+        .databaseResolution(DatabaseResolution.PREFIXED)
+        .primary("primary", localServer.getThriftConnectionUri(), READ_ONLY)
+        .federate(SECONDARY_METASTORE_NAME, remoteServer.getThriftConnectionUri(), REMOTE_DATABASE)
+        .build();
+
+    runWaggleDance(runner);
+    HiveMetaStoreClient proxy = getWaggleDanceClient();
+    ArrayList<ResourceUri> resourceUris = Lists
+        .newArrayList(new ResourceUri(ResourceType.JAR, "hdfs://path/to/my/jar/my.jar"));
+    Function localFunction = new Function("fn1", LOCAL_DATABASE, "com.hotels.hive.FN1", "hadoop", PrincipalType.USER, 0,
+        FunctionType.JAVA, resourceUris);
+    localServer.client().createFunction(localFunction);
+    Function remoteFunction = new Function("fn2", REMOTE_DATABASE, "com.hotels.hive.FN1", "hadoop", PrincipalType.USER,
+        0, FunctionType.JAVA, resourceUris);
+    remoteServer.client().createFunction(remoteFunction);
+
+    GetAllFunctionsResponse allFunctions = proxy.getAllFunctions();
+    List<Function> functions = allFunctions.getFunctions();
+    assertThat(functions.size(), is(2));
+    assertThat(functions.get(0).getFunctionName(), is("fn1"));
+    assertThat(functions.get(0).getDbName(), is(LOCAL_DATABASE));
```

Comment:
If primary also has a prefix, then this should return the prefixed version of the local database, right?

Reply:
yes